sms_sponsorship/controllers/main.py

```diff
@@ -142,20 +142,24 @@ def sms_change_child(self, child_request_id):
            auth='public', website=True, noindex=['robots', 'meta', 'header'])
     def step2_confirm_sponsorship(self, sponsorship_id=None, **kwargs):
         """ SMS step2 controller. Returns the sponsorship registration form."""
-        sponsorship = request.env['recurring.contract'].sudo().browse(
-            sponsorship_id)
-        if sponsorship.sms_request_id.state == 'step2' or \
-                sponsorship.state in ['active', 'waiting']:
-            # Sponsorship is already confirmed
-            return self.sms_registration_confirmation(sponsorship.id, **kwargs)
-        if sponsorship.state == "draft":
-            return self.make_response(
-                'recurring.contract',
-                model_id=sponsorship and sponsorship.id,
-                **kwargs
-            )
+        if sponsorship_id:
+            sponsorship = request.env['recurring.contract'].sudo().search([
+                ('id', '=', sponsorship_id)
+            ])
+            if sponsorship.sms_request_id.state == 'step2' or \
+                    sponsorship.state in ['active', 'waiting']:
+                # Sponsorship is already confirmed
+                return self.sms_registration_confirmation(sponsorship.id, **kwargs)
+            if sponsorship.state == "draft":
+                return self.make_response(
+                    'recurring.contract',
+                    model_id=sponsorship and sponsorship.id,
+                    **kwargs
+                )
+            else:
+                raise NotFound()
         else:
-            raise NotFound()
+            raise Exception("Please provide a sponsorship_id")
```

Comment:
raise NotFound() was better (for user interface), however you can add a line to log an error in the logs
```
import logging

_logger = logging.getLogger(__name__)

_logger.error("No sponsorship_id was given to the route")